### DISCUSSION TITLE
Too many slow layers above arcs

### DISCUSSION THREAD

**Discussion by JesseOfTheForest** [nicolai-wachenschwan/arc-overhang-prusaslicer-integration#35] [Q&A]
I'm printing with PETG currently, and the script slowed down all printing above the arcs for 17 layers. With PETG printing at the very slow rate it caused a number of extrusion problems and damaged the overall quality of the arcs.  What I would like to do is print the arcs at 1 speed, the first 2 or 3 layers above at a slightly faster speed, then return at least up to my defined bridging speeds in prusa.

Looking at the code I see 2 possible settings I could change, I'm wondering what is the best method that won't cause any other issues.
"CoolingSettingDetectionDistance"
"specialCoolingZdist"
Those 2 seem the most obvious and I will play around with them, but is there a better method? Are either of these settings likely to cause other problems if I adjust them incorrectly? Is it possible to increase the speed progressively over a few layers?

I think this information would also be useful in the readme, as I imagine this will be an issue for others.

**Comment by nicolai-wachenschwan** [comment]
Yeah, an improved documentation on the parameters is needed.
The special coolingZdist is exactly made for this purpose. adjust it as you like.
having it to low will increase warping.
If you print another material like petg you need to adjust a lot of settings for your machine.
Change a little and make test print. than change progressively.
Would be happy if you post your settings when you are happy with them!

Progressive speed change is not implemented yet.

**Reply by logiota** [reply]
is there a speed multiplier i can change ?

**Comment by nicolai-wachenschwan** [comment]
Yes, 2: one for infill and perimeter aboveArcsPerimeterPrintSpeed
